gcp_variant_transforms/testing/integration/run_tests_common.py

```diff
@@ -16,60 +16,65 @@
 
 It provides common functions and classes for both run_vcf_to_bq_tests
 (integration test script for vcf_to_bq pipeline) and run_preprocessor_tests
-(integration test script for vcf_to_bq preprocessor pipeline).
+(integration test script for vcf_to_bq_preprocess pipeline).
 """
 
-from abc import abstractmethod
-from typing import Dict, List  # pylint: disable=unused-import
 import argparse  # pylint: disable=unused-import
 import json
 import os
 import time
+from typing import Dict, List  # pylint: disable=unused-import
 
 from googleapiclient import discovery
+from oauth2client.client import GoogleCredentials
 
-DEFAULT_IMAGE_NAME = 'gcr.io/gcp-variant-transforms/gcp-variant-transforms'
+_DEFAULT_IMAGE_NAME = 'gcr.io/gcp-variant-transforms/gcp-variant-transforms'
+_DEFAULT_ZONES = ['us-west1-b']
 
 
 class TestCaseFailure(Exception):
   """Exception for failed test cases."""
   pass
 
 
-class TestCase(object):
-  """Test case that holds information to run in pipelines API."""
-
-  def __init__(self, test_name, project):
-    # type: (str, str) -> None
-    self._name = test_name
-    self._project = project
-
-  def get_name(self):
-    return self._name
-
-  def run(self, context):
-    service = discovery.build(
-        'genomics', 'v1alpha2', credentials=context.credentials)
-    # The following pylint hint is needed because `pipelines` is a method that
-    # is dynamically added to the returned `service` object above. See
-    # `googleapiclient.discovery.Resource._set_service_methods`.
-    # pylint: disable=no-member
-    request = service.pipelines().run(body=self._pipelines_api_request)
-
-    operation_name = request.execute()['name']
-    response = self._wait_for_operation_done(service, operation_name)
-    self._handle_failure(response)
-
-  def _wait_for_operation_done(self, service, operation_name):
-    """Waits until the operation ``operation_name`` of ``service`` is done."""
+class TestRunner(object):
+  """Runs the tests using pipelines API."""
+
+  def __init__(self, tests):
+    # type: (List[Union[PreprocessorTestCase, VcfToBQTestCase]]) -> None
+    self._tests = tests
+    self._service = discovery.build(
+        'genomics',
+        'v1alpha2',
+        credentials=GoogleCredentials.get_application_default())
+    self._operation_names = []
+    self._responses = []
```

Comment:
Just a suggestion: For class properties with non obvious types, I have sometime added type hints (e.g., [here](https://github.com/googlegenomics/gcp-variant-transforms/blob/1ff89befd1c566729c63d5500482288845ba673a/gcp_variant_transforms/libs/processed_variant.py#L86)) and I found it a useful practice. I thought maybe for this (and the previous property), it makes sense to add type hints as well, but up to you.